abl/util/lockfile.py

```diff
@@ -60,7 +60,8 @@ def __enter__(self):
             else:
                 fcntl.flock(self.file, lock_flags)
         except IOError, e:
-            if e[0] == errno.EAGAIN:
+            error_code = errno.EACCES if is_windows else errno.EAGAIN
```

Comment:
I think it makes more sense to check both EACCESS and EAGAIN because fcntl.flock's intention seems to be:
1- use bsd locks when available
2- use fcntl otherwise so the fcntl posix option seems to be the "default".

It seems hard to find a good predicate that would always work and do the right thing.

There's even the potential case where bsd locks' EWOULDBLOCK would be different from EAGAIN, on some old unix systems. See for instance: http://stackoverflow.com/questions/7003234/which-systems-define-eagain-and-ewouldblock-as-different-values

So to refine my answer I would check EAGAIN, EWOULDBLOCK and EACCESS and accept those all as a lock not being available, to cover all cases, without attempting to predicate on the OS. What's really important is that the lock call is the only thing covered by the try/catch that could produce an IOError.

Going even further I would rather have two codepaths one for mscrt (windows) and one for fcntl, because ultimately these are two different things and trying to homogenize error handling is the source of the issue with most ifdef infested OS layer code.

A little bit of duplication there might make things clearer and easier for people fixing code on the native platform they have access to.

```python
if is_windows:
   try:
        # msvcrt ... locking
   except IOError:
       # msvcrt error handling
else:
   try:
        # fcntl locking etc...
```